### PR TITLE
drivers: wifi: Send user timeout to FW

### PR DESCRIPTION
**Pull request by krish2718**
User can configure watchdog timer using Kconfig and this needs to be sent to FW. This is primarily needed for crowded channel where RPU can stay awake for more than 10s.

### CODE REVIEW

**Review by amar-nordic — APPROVED** [review]

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c [hunk 1]
```diff
@@ -149,6 +149,13 @@ enum nrf_wifi_status umac_cmd_init(struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx,
 	umac_cmd_data->tcp_ip_checksum_offload = 1;
 #endif /* CONFIG_NRF700X_TCP_IP_CHECKSUM_OFFLOAD */
 	umac_cmd_data->discon_timeout = CONFIG_NRF_WIFI_AP_DEAD_DETECT_TIMEOUT;
+#ifdef CONFIG_NRF_WIFI_RPU_RECOVERY
+	umac_cmd_data->watchdog_timer_val =
+		(CONFIG_NRF_WIFI_RPU_RECOVERY_PS_ACTIVE_TIMEOUT_MS) / 1000;
+#else
+	/* Disable watchdog */
+	umac_cmd_data->watchdog_timer_val = 0xFFFFFF;
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c:157]: Is this expected to be just 6 `F's` or a typo?

Comment (krish2718) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c:157]: Expected, it is a 24-bit counter.

**Review by sachinthegreen — COMMENTED** [review]
Also is the order of commits correct? (As in shouldn't the FW patch change come before the driver change)

**Review by sachinthegreen — APPROVED** [review]

**Comment by krish2718** [comment]
> Also is the order of commits correct? (As in shouldn't the FW patch change come before the driver change)

yes, both are independent commits.

**Review by bama-nordic — APPROVED** [review]

**Pull request merged**